src/test.rs

```diff
@@ -6,24 +6,50 @@ mod test {
     #[should_panic]
     fn http_get() {
         // make sure a valid url does not return nothing
-        assert_ne!(blindfold::http_get("https://api.github.com/repos/toptal/gitignore/\
-                                       contents/templates?ref=master"), "");
+        assert_ne!(
+            blindfold::http_get(
+                "https://api.github.com/repos/toptal/gitignore/\
+                                       contents/templates?ref=master"
+            ),
+            ""
+        );
         // make sure that an invalid url causes an error (ok because the repo url is hardcoded)
         panic!(blindfold::http_get("www.notarealsite/foo/bar"));
     }
 
+    #[test]
+    fn coalesece_globstars() {
+
+        let path_reduction_map: HashMap<&str, &str> = [
+            ("test/test", "test/test"),
+            ("test/**/test", "test/**/test"),
+            ("test/**/**/test", "test/**/test"),
+            ("test/**/**/**/test", "test/**/test"),
+            ("test/**/test/**/test", "test/**/test/**/test")
+        ].iter().cloned().collect();
```

Comment:
One last nitpick ;).  This can just be a plain array slice.  It can be turned into this:
```rust
let path_reduction_map = &[
...
];
```
which will be a lot more cleaner and also faster.

You might have to deference the loop values, though, to make the types work out.

Reply:
So just to clarify, this can be an array slice because it's defined at compile time. But if I were to dynamically insert elements, it would have to be a HashMap?